lib/lambda_ethereum_consensus/store/block_db.ex

```diff
@@ -10,31 +10,59 @@ defmodule LambdaEthereumConsensus.Store.BlockDb do
   @block_prefix "blockHash"
   @blockslot_prefix "blockSlot"
 
-  @spec store_block(SignedBeaconBlock.t(), Types.root()) :: :ok
-  def store_block(%SignedBeaconBlock{} = signed_block) do
-    store_block(signed_block, Ssz.hash_tree_root!(signed_block.message))
+  defmodule BlockInfo do
+    @moduledoc """
+    Signed beacon block accompanied with its root and its processing status.
+    Maps to what's saved on the blocks db.
+    """
+    @type block_status ::
+            :pending
+            | :invalid
+            | :processing
+            | :download
+            | :download_blobs
+            | :unknown
+            | :transitioned
+
+    @type t :: %__MODULE__{
+            root: Types.root(),
+            signed_block: Types.SignedBeaconBlock.t(),
+            status: block_status()
+          }
+    defstruct [:root, :signed_block, :status]
+
+    @spec from_block(SignedBeaconBlock.t(), block_status()) :: t()
+    def from_block(signed_block, status \\ :pending) do
+      {:ok, root} = Ssz.hash_tree_root(signed_block.message)
+      from_block(signed_block, root, status)
+    end
+
+    @spec from_block(SignedBeaconBlock.t(), Types.root(), block_status()) :: t()
+    def from_block(signed_block, root, status) do
+      %__MODULE__{root: root, signed_block: signed_block, status: status}
+    end
   end
 
-  def store_block(%SignedBeaconBlock{} = signed_block, block_root) do
-    {:ok, encoded_signed_block} = Ssz.to_ssz(signed_block)
+  def store_block(_root, block_info), do: store_block(block_info)
+
+  def store_block(%BlockInfo{} = block_info) do
```

Comment:
What do you think about renaming this into `store_block_info`? Calling `get_block` is not the same as calling `get_block_info` in `StoreBlocks`, maybe we can continue with that rule.

Reply:
Good idea! Done.